addons/mrp/models/mrp_bom.py

```diff
@@ -127,6 +127,8 @@ def onchange_product_tmpl_id(self):
                 self.product_id = False
             for line in self.bom_line_ids:
                 line.bom_product_template_attribute_value_ids = False
+            if self.product_tmpl_id.bom_count:  # add a reference to the bom if there is already a bom for this product
```

Comment:
There is a little issue: 
- Open a existant BoM
- Change the product_tmpl_id
- Rollback your change about the `product_tmpl_id`
- The `code` has been changed, it shouldn't

You shouldn't take in account the `self` in the `bom_count` (Use `bom_ids - self` maybe).